### PR TITLE
[api-documenter] Experimental doc plugin for Docusaurus

### PR DESCRIPTION
**Pull request by elliot-nelson**
## Summary

 - Rename the `doc-plugin-rush-stack` project to `doc-plugin-jekyll`, and adjust the current references in `generate-api-docs`.
 - Introduce a cloned-and-tweaked plugin `doc-plugin-docusaurus`, that produces the equivalent output for Docusaurus instead of Jekyll.
 - Minor tweaks to some tsdoc comments in `rush-lib`.

## Details

This PR is one step towards prepping for an experimental conversion of our documentation websites from Jekyll to Docusaurus.  This PR _doesn't_ change anything about the current output of the `generate-api-docs` repo-script, but you can replace "jekyll" with "docusaurus" in a branch and run it to produce the Docusaurus flavor instead.

(It's hard to make a truly generic API docs output, but I think these two projects make a good stab at their respective targets, and really aren't that rushstack-specific.  I've added initial READMEs for both written as if we were going to release them as official plugins, or at least samples, rather than just internal scripts.)

## How it was tested

Compare current rushstack site with the example output (*note* not intended to be a suggested replacement website yet; this just uses the output of the docusaurus plugin in an out-of-the-box Docusaurus website).

- [Executable.spawnSync on rushstack.io](https://rushstack.io/pages/api/node-core-library.executable.spawnsync/)
- [Executable.spawnSync on example docusaurus site](https://enelson-rushjs-docusaurus-mvp.netlify.app/pages/api/node-core-library.executable.spawnsync/)

### CODE REVIEW

File: common/config/rush/browser-approved-packages.json [hunk 1]
```diff
@@ -2,6 +2,10 @@
 {
   "$schema": "https://developer.microsoft.com/json-schemas/rush/v5/approved-packages.schema.json",
   "packages": [
+    {
+      "name": "doc-plugin-jekyll",
+      "allowedCategories": [ "libraries" ]
+    },
     {
```
Comment (octogonz) [inline common/config/rush/browser-approved-packages.json:9]: Move this to nonbrowser-approved-packages.json

Comment (octogonz) [reply common/config/rush/browser-approved-packages.json:9]: (To get Rush to sort+autoformat the files for you: after you move the JSON entry, delete `"libraries"` from your  array, then run `"rush update"`.  I've been wanting to improve this experience heheh...)

File: repo-scripts/doc-plugin-docusaurus/src/DocusaurusMarkdownFeature.ts [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+    const navFilePath: string = path.join(this.context.outputFolder, '..', 'api_nav.json');
+    const navFileContent: string = JSON.stringify(navigationFile, undefined, 2);
+
+    FileSystem.writeFile(navFilePath, navFileContent, { ensureFolderExists: true });
+  }
+
+  private _buildNavigation(parentNodes: INavigationNode[], parentApiItem: ApiItem): void {
+    for (const apiItem of parentApiItem.members) {
+      if (this._apiItemsWithPages.has(apiItem)) {
+        const label = apiItem.displayName;
```
Comment (octogonz) [inline repo-scripts/doc-plugin-docusaurus/src/DocusaurusMarkdownFeature.ts:77]: This project seems to be missing eslint

**Comment by octogonz** [comment]
LGTM!  I like the idea of keeping the Jekyll plugin around as a community sample.

**Review by octogonz — APPROVED** [review]

**Comment by elliot-nelson** [comment]
Closing for now -- the Docusuaurus plugin has moved into the rushstack-websites repo.

**Pull request closed without merging**